enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/readreplica/ReadReplicaGraphDatabase.java

```diff
@@ -36,14 +38,15 @@
 {
     public ReadReplicaGraphDatabase( File storeDir, Config config, Dependencies dependencies )
     {
-        this( storeDir, config, dependencies, new HazelcastDiscoveryServiceFactory() );
+        this( storeDir, config, dependencies, new HazelcastDiscoveryServiceFactory(), new MemberId( UUID.randomUUID() ) );
```

Comment:
For the edition module, it does. It's EnterpriseNeoServer that calls the shorter constructor, which ultimately ends up in the harness code. In the edition module we do in fact create and inject the MemberId.